common/isolationgroup/isolationgroupapi/global-api-handlers_test.go

```diff
@@ -0,0 +1,137 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package isolationgroupapi
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/uber/cadence/common/dynamicconfig"
+	"github.com/uber/cadence/common/log/loggerimpl"
+	"github.com/uber/cadence/common/types"
+)
+
+func TestUpdateGlobalState(t *testing.T) {
+
+	tests := map[string]struct {
+		in           types.UpdateGlobalIsolationGroupsRequest
+		dcAffordance func(client *dynamicconfig.MockClient)
+		expectedErr  error
+	}{
+		"updating normal value": {
+			in: types.UpdateGlobalIsolationGroupsRequest{IsolationGroups: types.IsolationGroupConfiguration{
+				"zone-1": {Name: "zone-1", State: types.IsolationGroupStateHealthy},
+				"zone-2": {Name: "zone-2", State: types.IsolationGroupStateDrained},
+			}},
+			dcAffordance: func(client *dynamicconfig.MockClient) {
+				client.EXPECT().UpdateValue(
+					dynamicconfig.DefaultIsolationGroupConfigStoreManagerGlobalMapping,
+					gomock.Any(), // covering the mapping in the mapper unit-test instead
+				)
+			},
+		},
```

Comment:
nit: an error case

Reply:
tricky... I am not sure I can trigger an error case given the type system and the only thing that can trigger an error is the json serialization